godel_surface_detection/src/nodes/robot_auto_blending.cpp

```diff
@@ -13,87 +11,112 @@
+  ros::ServiceClient sim_client_ = nh.serviceClient<godel_msgs::SelectMotionPlan>(SELECT_MOTION_PLAN_SERVICE);
+
   while (ros::ok())
   {
+    if (serviceCall_failure > MAX_SERVICECALL_FAILURE )
+    {
+      ROS_ERROR_STREAM("Service Call failures exceed the MAX_SERVICECALL_FAILURE");
+      return -1;
+    }
     //Get parameters for surface detection
-    godel_msgs::SurfaceBlendingParameters param_srv;
     param_srv.request.action = param_srv.request.GET_CURRENT_PARAMETERS;
-    ros::ServiceClient param_client = nh.serviceClient<godel_msgs::SurfaceBlendingParameters>(SURFACE_BLENDING_PARAMETERS);
-    std::vector<std::string> plan_names;
-
-    if (param_client.call(param_srv))
+    if (!param_client.call(param_srv))
     {
-      //Scan and detect surface
-      godel_msgs::SurfaceDetection surfacedetection_srv;
-      surfacedetection_srv.request.action = 3;
-      surfacedetection_srv.request.use_default_parameters = false;
-      surfacedetection_srv.request.robot_scan = param_srv.response.robot_scan;
-      surfacedetection_srv.request.surface_detection = param_srv.response.surface_detection;
-      ros::ServiceClient surface_client_ = nh.serviceClient<godel_msgs::SurfaceDetection>(SURFACE_DETECTION_SERVICE);
-
-      if (!surface_client_.call(surfacedetection_srv))
-      {
-        ROS_WARN_STREAM("Unable to call surface detection service");
-      }
+      ROS_WARN_STREAM("Unable to call surface_blending_parameters service");
+      serviceCall_failure++;
+      continue;
+    }
 
-      //Select all surface
-      godel_msgs::SelectSurface::Request req;
-      godel_msgs::SelectSurface::Response res;
-      req.action = req.SELECT_ALL;
-      ros::ServiceClient select_surface_client_ = nh.serviceClient<godel_msgs::SelectSurface>(SELECT_SURFACE_SERVICE);
-      bool succeeded = select_surface_client_.call(req, res);
+    //Scan and detect surface
+    surfacedetection_srv.request.action = 3;
+    surfacedetection_srv.request.use_default_parameters = false;
+    surfacedetection_srv.request.robot_scan = param_srv.response.robot_scan;
+    surfacedetection_srv.request.surface_detection = param_srv.response.surface_detection;
+    if (!surface_client_.call(surfacedetection_srv))
+    {
+      ROS_WARN_STREAM("Unable to call surface detection service");
+      serviceCall_failure++;
+      continue;
+    }
 
-      //Generate path
-      godel_msgs::ProcessPlanning process_plan;
-      process_plan.request.use_default_parameters = false;
-      process_plan.request.params = param_srv.response.blending_plan;
-      process_plan.request.scan_params =param_srv.response.scan_plan;
-      process_plan.request.action = 2;
-      ros::ServiceClient process_plan_client_ = nh.serviceClient<godel_msgs::ProcessPlanning>(PROCESS_PATH_SERVICE);
+    //Select all surface
+    req.action = req.SELECT_ALL;
+    if (!select_surface_client_.call(req, res))
+    {
+      ROS_WARN_STREAM("Unable to call select_surface service");
```

Comment:
On these ROS_WARN statements you could use the global string variable that has the service name in it.  For instance ROS_WARN_STREAM("Unable to call the "<<SELECT_SURFACE_SERVICE<<" service").  This would make it easier to change the value of that variable since you won't have to update every single ROS_STREAM statement that refers to it.